src/main/java/com/coreos/jetcd/internal/impl/KVImpl.java

```diff
@@ -56,9 +59,14 @@
         .setPrevKv(option.getPrevKV())
         .build();
 
-    return Util.toCompletableFuture(
+    return Util.toCompletableFutureWithRetry(
         stub.put(request),
         PutResponse::new,
+        () -> {
+          stub = connectionManager.newStubWithNewToken(KVGrpc::newFutureStub);
+          return stub.put(request);
+        },
+        Util::retry,
```

Comment:
Even though retry twice won't violate the correctness of the code, but we should not allow retry twice. If there is ongoing retry, the second retry should wait until first one finishes and use that token from first retry to reissue the request.